### PR TITLE
Header Refactoring

### PR DESCRIPTION
**Pull request by dmccoystephenson**
# Description
The bsmfilter.hpp file defines 4 classes at this time. I decided to spend some time tonight splitting this file (and the associated .cpp file) up in order to make the PPM easier to work on. These changes are from my fork of the project.

The bsmfilter.hpp and bsmfilter.cpp files have been split up into 8 files, listed below:
- bsmHandler.hpp
- bsm.hpp
- idRedactor.hpp
- velocityFilter.hpp
- bsmHandler.cpp
- bsm.cpp
- idRedactor.cpp
- velocityFilter.hpp

This has been marked as a draft PR for now until functionality is verified with docker-compose and in GCP.
## Acceptance Testing Criteria
- [x] Unit tests pass in dev container.
- [x] PPM starts up successfully with docker-compose
- [x] PPM starts up successfully in GCP k8s cluster.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,69 @@
+{
+    "files.associations": {
```
Comment (payneBrandon) [inline .vscode/settings.json:2]: Is all of this needed? Seems odd we'd need to specify all these file types to associate to cpp.

Comment (dmccoystephenson) [reply .vscode/settings.json:2]: I don't think this is needed as it pertains to the IDE, not the project itself. This has been removed and the unit tests are still passing.

File: src/idRedactor.cpp [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    // setup random number generator.
+    std::random_device rd;
+    rgen_ = std::mt19937{ rd() };
+    dist_ = std::uniform_int_distribution<uint32_t>{ 0, std::numeric_limits<uint32_t>::max() };
+}
+
+IdRedactor::IdRedactor( const ConfigMap& conf ) :
+    IdRedactor{}
+{
+    // TODO: The redaction value is deprecated (it is randomly assigned now).
```
Comment (payneBrandon) [inline src/idRedactor.cpp:17]: I assume this todo was here previously?

Comment (dmccoystephenson) [reply src/idRedactor.cpp:17]: Yes, it was.

**Review by payneBrandon — APPROVED** [review]
Looks good to me!

**Pull request merged**